tests/document/instance.py

```diff
@@ -3186,8 +3181,7 @@ class BlogPost(Document):
 
         blog.update(push__tags__0=["mongodb", "code"])
         blog.reload()
-        self.assertEqual(blog.tags[0], "mongodb")
-        self.assertEqual(blog.tags[2], "python")
+        self.assertEqual(blog.tags, ['mongodb', 'code', 'python'])
```

Comment:
I disagree with you, in the query transform, there is a code that checks value is list or not. If its list, then return each modifier.
<pre>
+        elif op == 'push' and isinstance(value, list):
 +            if position is not None:
 +                value = {key: {'$each': value, '$position': position}}
</pre>
This behavior was same in addtoset also, then it has to be list even if single element will be pushed or we could change the behavior for the singular value. What do you suggest?